### DISCUSSION TITLE
Bug: Query react supsense not working in example

### DISCUSSION THREAD

**Discussion by BennyAlex** [pmndrs/jotai#1932] [Q&A]
Hello,

the react supsense example for jotai query https://codesandbox.io/s/joer59 does not work.
When clickign to id 0 a error is thrown.

![image](https://github.com/pmndrs/jotai/assets/17402724/bc2ba6eb-f37f-4a86-a3a9-f41897526c5c)

@dai-shi 
Could you also explain why and when to use jotai-query instead of just using tanstack query?
You can even pass atom values as query keys, so why should I use atomWithQuery?

Thanks for your help.

**Comment by dai-shi** [comment]
> https://codesandbox.io/s/joer59 does not work.

Can anyone look into it?

> when to use jotai-query instead of just using tanstack query?

You would only need it when you want to derive atoms combining an atom from tanstack query and others from somewhere else. (all in atom definitions, without using hooks.)
